src/compiler/checker.ts

```diff
@@ -18159,6 +18165,25 @@ namespace ts {
             return suggestion && symbolName(suggestion);
         }
 
+        function getSuggestionsForNonexistentIndexSignature(objectType: Type): string | undefined {
+            let suggestions: string | undefined;
+            const props = [
+                getPropertyOfObjectType(objectType, <__String>"get"),
+                getPropertyOfObjectType(objectType, <__String>"set")
+            ];
+
+            for (const prop of props) {

```

Comment:
I think that this is a little too general. You really only want to recommend `"set"` when the element access is on the left hand side of some mutating operator (e.g. `=`, `+=`, `++`, etc.). You want to try to recommend `"get"` otherwise.